src/features/Sync/SyncSlice.ts

```diff
@@ -544,51 +710,55 @@ function* syncTemperatureBreaches({ payload: { temperatureBreachUrl }}: SyncTemp
 }
 
 function* syncAll({ payload: { loginUrl, sensorUrl, temperatureLogUrl, temperatureBreachUrl, username, password }}: SyncAllAction): SagaIterator {
+    yield put(SyncAction.updateIsSyncing(true));
     yield put(SyncAction.authenticate(loginUrl, username, password))
-    yield delay(MILLISECONDS.TEN_SECONDS);
-    yield put(SyncAction.syncSensors(sensorUrl));
-    yield put(SyncAction.syncTemperatureLogs(temperatureLogUrl));
-    yield put(SyncAction.syncTemperatureBreaches(temperatureBreachUrl));
-}
-
-function* fetchAll(): SagaIterator {
-    const DependencyLocator = yield getContext(DEPENDENCY.LOCATOR);
-    const syncOutManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_OUT_MANAGER);
-    try {
-        const loginUrl = yield call(syncOutManager.getLoginUrl);
-        const sensorUrl = yield call(syncOutManager.getSensorUrl);
-        const temperatureLogUrl = yield call(syncOutManager.getTemperatureLogUrl);
-        const temperatureBreachUrl = yield call(syncOutManager.getTemperatureBreachUrl);
-        const username = yield call(syncOutManager.getUsername);
-        const password = yield call(syncOutManager.getPassword);
-
-        const syncConfig = {
-            loginUrl,
-            sensorUrl,
-            temperatureLogUrl,
-            temperatureBreachUrl,
-            username,
-            password,
-        };
-
-        yield put(SyncAction.fetchAllSuccess(syncConfig));
-    } catch (error) {
-        // eslint-disable-next-line no-console
-        console.log(error.message);
-        // TODO: add logic for handling failed pushes
+    const authenticateResult = yield take([SyncAction.authenticateSuccess, SyncAction.authenticateFailure]);
+    if (authenticateResult.type === SyncAction.authenticateSuccess.type) {
+        yield put(SyncAction.syncSensors(sensorUrl));
+        yield take([SyncAction.syncSensorsSuccess, SyncAction.syncSensorsFailure]);
+        yield put(SyncAction.syncTemperatureLogs(temperatureLogUrl));
+        yield take([SyncAction.syncSensorsSuccess, SyncAction.syncTemperatureLogsFailure]);
+        yield put(SyncAction.syncTemperatureBreaches(temperatureBreachUrl));
+        yield take([SyncAction.syncTemperatureBreachesSuccess, SyncAction.syncTemperatureBreachesFailure]);
+    }
+    yield put(SyncAction.updateIsSyncing(false));
+}
+
+function* startSyncScheduler(): SagaIterator {
+    while (true) {
+        const loginUrl = yield select(getLoginUrl);
+        const sensorUrl = yield select(getSensorUrl);
+        const temperatureLogUrl = yield select(getTemperatureLogUrl);
+        const temperatureBreachUrl = yield select(getTemperatureBreachUrl);
+        const username = yield select(getUsername);
+        const password = yield select(getPassword);
```

Comment:
I think avoiding use of the store when you can is good in the sagas. The saga is now dependent on when the store is rehydrated, which is dependent on the implementation in the UI and the database. 

If we were to fetch these details from the db (with a `getConfig` method 🤣 ), we are only dependent on the db

Reply:
Resolved by #91.